mp/src/game/shared/momentum/util/engine_patch.cpp

```diff
@@ -0,0 +1,243 @@
+        20,
+        false,
+        (float)-1
+    }
+};
+#endif //WIN32
+
+void CEnginePatch::ApplyAll()
+{
+#if !defined (OSX) // No OSX patches yet
+    int err;
+
+    for (int i = 0; i < sizeof(g_EnginePatches) / sizeof(*g_EnginePatches); i++)
+    {
+        err = g_EnginePatches[i].ApplyPatch();
+        switch (err)
+        {
+        case 1:
+            Warning("Engine patch \"%s\" FAILED: Could not override memory protection\n", g_EnginePatches[i].getName());
+            break;
+        case 2:
+            Warning("Engine patch \"%s\" FAILED: Could not find signature\n", g_EnginePatches[i].getName());
+        }
+    }
+#endif // (OSX)
+}
+
+// Apply a patch to engine memory
+int CEnginePatch::ApplyPatch()
+{
+#ifdef _WIN32
+
+    uintptr_t addr = reinterpret_cast<uintptr_t>(CEngineBinary::FindPattern(m_pPatchSignature, m_pMask, m_iOffset));
+
+    if (addr)
+    {
+        auto pMemory = m_bImmediate ? (uintptr_t*)addr : *reinterpret_cast<uintptr_t**>(addr);
+
+        // Memory is write-protected so it needs to be lifted before the patch is applied
+        // 0x40 is read,write,execute
+        // https://docs.microsoft.com/en-us/windows/win32/memory/memory-protection-constants
+        unsigned long iOldProtection, iNewProtection = 0x40;
+
+        if (VirtualProtect(pMemory, m_iPatchLength, iNewProtection, &iOldProtection))
+        {
+            Q_memcpy(pMemory, m_pPatch, m_iPatchLength);
+
+            // Restore old protections
+            VirtualProtect(pMemory, m_iPatchLength, iOldProtection, &iNewProtection);
+
+            return 0;
+        }
+        return 1;
+    }
+    return 2;
+
+#elif defined __linux__
+
+    void* addr = CEngineBinary::FindPattern(m_pPatchSignature, m_pMask, m_iOffset);
+
+    if (addr)
+    {
+        auto pMemory = m_bImmediate ? (uintptr_t*)addr : *(uintptr_t**)(addr);
+
+        // Memory is write-protected so it needs to be lifted before the patch is applied
+        // http://man7.org/linux/man-pages/man2/mprotect.2.html
+        if (!mprotect(pMemory, m_iPatchLength, PROT_READ | PROT_WRITE | PROT_EXEC))
+        {
+            Q_memcpy(pMemory, m_pPatch, m_iPatchLength);
+
+            // Restore old protections
+            mprotect(pMemory, m_iPatchLength, PROT_READ | PROT_EXEC);
+
+            return 0;
+        }
+        return 1;
+    }
+    return 2;
+
+#endif //WIN32
+}
+
+// Constructors
+CEnginePatch::CEnginePatch(const char* name, unsigned char* signature, char* mask, size_t offset, bool immediate)
+{
+    m_sName = name;
+    m_pPatchSignature = signature;
+    m_pMask = mask;
+    m_iOffset = offset;
+    m_iPatchLength = 0;
+    m_bImmediate = immediate;
+    m_pPatch = new unsigned char;
+}
+
+CEnginePatch::CEnginePatch(const char* name, unsigned char* signature, char* mask, size_t offset, bool immediate, int value)
+    : CEnginePatch(name, signature, mask, offset, immediate)
+{
+    m_iPatchLength = sizeof(value);
+    Q_memcpy(m_pPatch, &value, m_iPatchLength);
```

Comment:
It looks to me like `m_pPatch` is assigned `new unsigned char` which can only store 1 byte, yet you're copying 4 bytes into it.

Reply:
Never noticed issues but that is indeed unhealthy